analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/SymbolicExecution/CollectorTestCheck.cs

```diff
@@ -64,9 +66,21 @@ public void ValidateTag(string tag, Action<SymbolicValue> action)
             var context = tags.Single(x => x.Name == tag).Context;
             var invocation = (IInvocationOperation)context.Operation.Instance;
             invocation.Arguments.Should().HaveCount(2, "Asserted argument is expected in Tag(..) invocation");
-            var symbol = ((ILocalReferenceOperation)((IConversionOperation)invocation.Arguments[1].Value).Operand).Local;
+            var symbol = ParameterOrLocalSymbol(((IConversionOperation)invocation.Arguments[1].Value).Operand);
             var value = context.State[symbol];
             action(value);
         }
+
+        private static ISymbol ParameterOrLocalSymbol(IOperation operation)
```

Comment:
Do we need to reimplement this? I've suggested extracting this as an extension method so we can use it here. And add proper UTs to that extension method.